infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/EventLogger.java

```diff
@@ -88,6 +88,32 @@ public void syncCompleted() {
     info("Syncing completed", Color.GREEN);
   }
 
+  public void headNoMoreOptimisticWhileSyncing() {
+    info("Execution Client caught up Beacon chain while syncing", Color.YELLOW);
+  }
+
+  public void headTurnedOptimisticWhileSyncing() {
+    info("Execution Client fallen behind Beacon chain while syncing", Color.YELLOW);
+  }
+
+  public void headTurnedOptimisticWhileInSync() {
+    info(
+        "Unable to verify head against Execution Client. Waiting the Execution Client to be back in sync",
+        Color.YELLOW);
+  }
+
+  public void syncCompletedWhileHeadIsOptimistic() {
+    info("Beacon chain syncing complete, waiting for Execution Client", Color.YELLOW);
+  }
+
+  public void executionClientIsOffline() {
+    info("Execution Client is offline. See teku logs for more details", Color.RED);
```

Comment:
Wil we actually have more details in the teku logs? I wonder if it's worth including the initial exception that caused the failure here.

I'd also log this as an `error` level.

Reply:
right, at least not always.